hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1691,24 +1682,39 @@ func reconcileCAPICluster(cluster *capiv1.Cluster, hcluster *hyperv1.HostedClust
 	cluster.Annotations = map[string]string{
 		hostedClusterAnnotation: client.ObjectKeyFromObject(hcluster).String(),
 	}
-	gvk, err := apiutil.GVKForObject(infraCR, api.Scheme)
-	if err != nil {
-		return err
-	}
-	cluster.Spec = capiv1.ClusterSpec{
-		ControlPlaneEndpoint: capiv1.APIEndpoint{},
-		ControlPlaneRef: &corev1.ObjectReference{
-			APIVersion: "hypershift.openshift.io/v1alpha1",
-			Kind:       "HostedControlPlane",
-			Namespace:  hcp.Namespace,
-			Name:       hcp.Name,
-		},
-		InfrastructureRef: &corev1.ObjectReference{
-			APIVersion: gvk.GroupVersion().String(),
-			Kind:       gvk.Kind,
-			Namespace:  infraCR.GetNamespace(),
-			Name:       infraCR.GetName(),
-		},
+	var gvk schema.GroupVersionKind
+	var err error
+	if infraCR != nil { //non platform none
+		gvk, err = apiutil.GVKForObject(infraCR, api.Scheme)
+		if err != nil {
+			return err
+		}
+		cluster.Spec = capiv1.ClusterSpec{
+			ControlPlaneEndpoint: capiv1.APIEndpoint{},
+			ControlPlaneRef: &corev1.ObjectReference{
+				APIVersion: "hypershift.openshift.io/v1alpha1",
+				Kind:       "HostedControlPlane",
+				Namespace:  hcp.Namespace,
+				Name:       hcp.Name,
+			},
+			InfrastructureRef: &corev1.ObjectReference{
+				APIVersion: gvk.GroupVersion().String(),
+				Kind:       gvk.Kind,
+				Namespace:  infraCR.GetNamespace(),
+				Name:       infraCR.GetName(),
+			},
+		}
+	} else { //platform none
+		cluster.Spec = capiv1.ClusterSpec{
```

Comment:
should probably not create a cluster at all. It will never reconcile successfully without an infra CR.

Reply:
```
if infra == nil {
 return early
}

Otherwise instantiate Cluster
```

Reply:
sorry I was missreading before. I think the most readable way would be:

```
	cluster.Spec = capiv1.ClusterSpec{
		ControlPlaneEndpoint: capiv1.APIEndpoint{},
		ControlPlaneRef: &corev1.ObjectReference{
			APIVersion: "hypershift.openshift.io/v1alpha1",
			Kind:       "HostedControlPlane",
			Namespace:  hcp.Namespace,
			Name:       hcp.Name,
		},
		InfrastructureRef: &corev1.ObjectReference{},
	}

if infra != nil {
 cluster.Spec.InfrastructureRef= &corev1.ObjectReference{
			APIVersion: gvk.GroupVersion().String(),
			Kind:       gvk.Kind,
			Namespace:  infraCR.GetNamespace(),
			Name:       infraCR.GetName(),
		},
}

```

We should pobably not create the CR at all, i.e don't call
```
	capiCluster := controlplaneoperator.CAPICluster(controlPlaneNamespace.Name, hcluster.Spec.InfraID)
	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, capiCluster, func() error {
		return reconcileCAPICluster(capiCluster, hcluster, hcp, infraCR)
	})
```
We can discuss and do that in a different PR though

Reply:
sorry but I don't know what you want to say. Let's talk tomorrow just to discuss ok?